### PR TITLE
fix(core): Double prepare on lex unit for ctas with union(#226)

### LINKED ISSUE #226: bug: Table not created, prompt Table 't2' already exists
### Describe the problem
```
1、create database syw_mtr;
2、use syw_mtr;
3、CREATE TABLE t1 (f1 VARCHAR(255) CHARACTER SET utf8)engine=stonedb;
4、CREATE TABLE t2 AS SELECT LEFT(f1,171) AS f2 FROM t1 UNION SELECT LEFT(f1,171) AS f2 FROM t1;
ERROR 1050 (42S01): Table 't2' already exists
```
### Expected behavior
```
1. No error message is displayed, and T2 can be created normally
2. note：Verify innoDB, normal without any error
```
### How To Reproduce
```
1、create database syw_mtr;
2、use syw_mtr;
3、CREATE TABLE t1 (f1 VARCHAR(255) CHARACTER SET utf8)engine=stonedb;
4、CREATE TABLE t2 AS SELECT LEFT(f1,171) AS f2 FROM t1 UNION SELECT LEFT(f1,171) AS f2 FROM t1;
```
### Environment
```
1. StoneDB for mysql5.7  (release)
2. Ubuntu 20.04.4
```
### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

### PR DESCRIPTION
**Pull request by xiaoguangye**
## Summary about this PR

Issue Number: close #226

## Tests Check List

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [ ] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Comment by andyli029** [comment]
Fix the commit log.

**Review by isredstar — APPROVED** [review]
LGTM

File: mysql-test/suite/tianmu/t/issue226.test [hunk 1]
```diff
@@ -0,0 +1,5 @@
+use test;
+CREATE TABLE t1 (f1 VARCHAR(255) CHARACTER SET utf8) ENGINE=TIANMU;
+CREATE TABLE t2 AS SELECT LEFT(f1,171) AS f2 FROM t1 UNION SELECT LEFT(f1,171) AS f2 FROM t1;
```
Comment (hustjieke) [inline mysql-test/suite/tianmu/t/issue226.test:3]: Good job!
It may be better to insert some data here to test  if`t2` is available or not.

Comment (xiaoguangye) [reply mysql-test/suite/tianmu/t/issue226.test:3]: Yeah, i've done that. Let's review again.

**Review by hustjieke — APPROVED** [review]
LGTM

**Pull request merged**